### PR TITLE
feat: remove Angular Compatibility Compiler (ngcc)

### PR DESCRIPTION
**Pull request by alan-agius4**
This commit removes the NGCC code and all the related infra setup required to support it.

BREAKING CHANGE: Angular Compatibility Compiler (ngcc) has been removed and as a result Angular View Engine libraries will no longer work

### CODE REVIEW

**Review by alxhub — APPROVED** [review]
Thanks, ngcc ❤️ but you will not be missed

**Review by josephperrott — APPROVED** [review]
LGTM

**Review by twerske — APPROVED** [review]
lgtm on the docs changes

**Review by alxhub — APPROVED** [review]
Reviewed-for: global-approval

**Review by alxhub — APPROVED** [review]
Reviewed-for: global-approvers

**Review by pkozlowski-opensource — APPROVED** [review]
LGTM

Reviewed-for: public-api

File: packages/compiler-cli/package.json [hunk 1]
```diff
@@ -58,7 +47,6 @@
     "reflect-metadata": "^0.1.2",
     "chokidar": "^3.0.0",
     "convert-source-map": "^1.5.1",
-    "dependency-graph": "^0.11.0",
     "magic-string": "^0.27.0",
```
Comment (JoostK) [inline packages/compiler-cli/package.json:50]: `magic-string` was only used in ngcc.

**Review by JoostK — APPROVED** [review]

File: packages/compiler-cli/package.json (hunk 1 above)
Comment (alan-agius4) [reply packages/compiler-cli/package.json]: good catch, removed

**Review by atscott — APPROVED** [review]

**Review by jessicajaniuk — APPROVED** [review]
reviewed-for: public-api, code-ownership

**Comment by alxhub** [comment]
This PR was merged into the repository by commit 48aa96ea13ebfadf2f6b13516c7702dae740a7be.

**Pull request closed without merging**